vars/githubWorkflowRun.groovy

```diff
@@ -0,0 +1,109 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+/**
+
+  Triggers workflow run on Github actions.
+
+  githubWorkflowRun( args map )
+
+  Args:
+    repo - repositiry owner and name 
+    workflow - workflow file name
+    ref - reference ( branch, tag or hash)
+    parameters - map with parameters to pass to the workflow 
+
+  Returs:
+    runInfo - information about
+
+  Example:
+     script {
+       Map args = [
+           repo: "owner/repository",
+           workflow: "build.yml", 
+           ref: "main",
+           parameters: [
+             path: "filebeat"
+           ],
+           credentialsId: "github-workflow-token",
+           ghVersion: "2.1.0"]
+       int runId = githubWorkflowRun.triggerGithubActionsWorkflow(args)
+       Map runInfo = githubWorkflowRun.getWorkflowRun(args + [runId: runId]) 
+   }
+*/
+
+import groovy.json.JsonSlurperClassic 
+
+def call(Map args = [:]) {
+    try {
```

Comment:
Is it needed to trap the error rather than letting the step to fail?

Reply:
removed this trap